substrate/frame/election-provider-multi-phase/test-staking-e2e/src/mock.rs

```diff
@@ -818,9 +815,7 @@ pub(crate) fn add_slash(who: &AccountId) {
 // Slashes enough validators to cross the `Staking::OffendingValidatorsThreshold`.
 pub(crate) fn slash_through_offending_threshold() {
```

Comment:
what does this do since we removed `OffendingValidatorsThreshold`?

Reply:
This function is used only [here[(https://github.com/paritytech/polkadot-sdk/blob/8ee4042c3bc86d4232e51a9bdb5ec1ae5ca444ea/substrate/frame/election-provider-multi-phase/test-staking-e2e/src/lib.rs#L143). Looks like a unit test for a specific regression. If an era is forced too soon the election fails. Should be irrelevant since we are removing the logic which forces a new era. 

But still is it a good idea to remove this test?

Reply:
I didn't notice that the test is failing. I've fixed the era checks so that it passes and I believe there is no benefit from the test anymore.

Reply:
@gpestana as we are removing force new era can we also remove [this test](https://github.com/paritytech/polkadot-sdk/blob/0a39a156756c9a3a3f81725094f20d0c9ac64f80/substrate/frame/election-provider-multi-phase/test-staking-e2e/src/lib.rs#L143)?

Reply:
The function now slashes 1/2 of the active set and is used in [this test](https://github.com/paritytech/polkadot-sdk/pull/2226/files#diff-961455df41e02f8ec6d63e469f4b5b684fdd550b310654bb8967c71b8c15e3baR146)